Sources/SPTDataLoaderSwift/Request+Concurrency.swift

```diff
@@ -0,0 +1,120 @@
+// Copyright 2015-2022 Spotify AB
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if compiler(>=5.5.2) && canImport(_Concurrency)
+
+import Foundation
+
+@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+public extension Request {
+    func task() -> ResponseTask<Void> {
+        return ResponseTask(request: self)
+    }
+
+    func dataTask() -> ResponseTask<Data> {
+        return ResponseTask(request: self)
+    }
+
+    func decodableTask<Value: Decodable>(decoder: ResponseDecoder = JSONDecoder()) -> ResponseTask<Value> {
+        return ResponseTask(request: self, decoder: decoder)
+    }
+
+    func jsonTask(options: JSONSerialization.ReadingOptions = []) -> ResponseTask<Any> {
+        return ResponseTask(request: self, options: options)
+    }
+
+    func serializableTask<Serializer: ResponseSerializer>(serializer: Serializer) -> ResponseTask<Serializer.Output> {
+        return ResponseTask(request: self, serializer: serializer)
+    }
+}
+
+// MARK: -
+
+@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+public struct ResponseTask<Value> {
+    private let task: Task<Response<Value, Error>, Never>
+
+    fileprivate init(
+        request: Request,
+        continuation: @escaping (CheckedContinuation<Response<Value, Error>, Never>) -> Void
+    ) {
+        self.task = Task {
+            await withTaskCancellationHandler(
```

Comment:
can we limit the use of `await/async` to iOS 15 only? Even though there is a backward compatibility library up to iOS 13. It is not clear how well it performs or what bugs might be covering under the surface at the moment and it would be much safer to wait a bit longer to use it

Reply:
Yeah, or make this a separate optional library

Reply:
Alright, I've just pushed a commit to limit them to the latest OS versions.